test/integration-test/src/utils.rs

```diff
@@ -0,0 +1,61 @@
+//! Utilities to run tests
+
+use std::{
+    ffi::CString,
+    os::fd::RawFd,
+    sync::atomic::{AtomicU64, Ordering},
+};
+
+use aya::netlink_set_link_up;
+use libc::if_nametoindex;
+use netns_rs::{get_from_current_thread, NetNs};
+
+pub struct Netns {
+    pub name: String,
+    pub old_ns: NetNs,
+    pub ns: Option<NetNs>,
+}
+
+impl Netns {
+    pub fn new() -> Self {
+        static COUNTER: AtomicU64 = AtomicU64::new(0);
+        let pid = std::process::id();
+        let name = format!("aya-test-{pid}-{}", COUNTER.fetch_add(1, Ordering::Relaxed));
+
+        // Create and enter netns
+        let ns = NetNs::new(&name).unwrap_or_else(|e| panic!("Failed to create netns {name}: {e}"));
+        ns.enter().expect("Failed to enter network namespace");
+        let netns = Self {
+            old_ns: get_from_current_thread().expect("Failed to get current netns"),
+            ns: Some(ns),
+            name,
+        };
+
+        // By default, the loopback in a new netns is down. Set it up.
+        let lo = CString::new("lo").unwrap();
+        unsafe {
+            let idx = if_nametoindex(lo.as_ptr());
+            if idx == 0 {
+                panic!("Interface `lo` not found in netns {}", netns.name);
+            }
+            netlink_set_link_up(idx as RawFd)
```

Comment:
RawFd is wrong here

Reply:
Why so? I copied this `as RawFd` from other uses of `if_nametoindex` in aya: see [`Xdp::attach`](https://github.com/aya-rs/aya/blob/main/aya/src/programs/xdp.rs#L99), that calls [`Xdp::attach_to_ifindex`](https://github.com/aya-rs/aya/blob/main/aya/src/programs/xdp.rs#L120) (uses `as RawFd`) which then calls [`aya::sys::netlink_set_xdp_fd`](https://github.com/aya-rs/aya/blob/main/aya/src/sys/netlink.rs#L26).

So if `RawFd` is wrong here, it's also wrong in the place I mentioned.

Reply:
I don't know about those other cites, but I think we can agree that an interface index is not a file descriptor?

Reply:
> I don't know about those other cites, but I think we can agree that an interface index is not a file descriptor?

Huh the sun hit hard, I didn't even realize this. Indeed that other use is wrong. Changing to `as i32`.